client/src/test/java/io/moquette/spi/impl/security/AESTest.java

```diff
@@ -0,0 +1,148 @@
+package io.moquette.spi.impl.security;
+
+import com.diffblue.deeptestutils.mock.DTUMemberMatcher;
+import org.junit.Test;
+import org.junit.Assert;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import java.lang.reflect.Method;
+import javax.crypto.Cipher;
+
+import static org.mockito.AdditionalMatchers.or;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Matchers.isNull;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.management.*"})
+public class AESTest {
+
+    @PrepareForTest({Cipher.class, AES.class})
+    @Test
+    public void testAESEncrypt() throws Exception {
+        PowerMockito.mockStatic(Cipher.class);
+
+        Cipher cipher = PowerMockito.mock(Cipher.class);
+        Method getInstanceMethod =
+            DTUMemberMatcher.method(Cipher.class, "getInstance", String.class);
+        PowerMockito.doReturn(cipher)
+            .when(Cipher.class, getInstanceMethod)
+            .withArguments(or(isA(String.class), isNull(String.class)));
+
+        Assert.assertNull(AES.AESEncrypt("", "0x000x110x220x33"));
```

Comment:
I think all these tests would work if you just remove all of the mocking (you will need to change the expected values though).